libs/sdk-core/src/breez_services.rs

```diff
@@ -2043,7 +2043,7 @@ pub(crate) mod tests {
 
         let received = breez_services
             .list_payments(ListPaymentsRequest {
-                filter: PaymentTypeFilter::Received,
+                filters: vec![PaymentTypeFilter::Received],
```

Comment:
If the user passes empty array here the underline storage will return sql error (where clause is empty).
So we need to validate the input here.
Alternatively how about considering empty array as `All` and remove the `All` from the enum variants? After all the array is named `filters` and if nothing to filter then it might make sense to return all items.